### ISSUE TITLE
Lint/RedundantCopDisableDirective improvement

### ISSUE THREAD

**Issue by sihu** [rubocop/rubocop-rspec#1550]
Dear Robocop-rspec team

In our case the redundant_cop_disable_directive.rb crashed on line 308:

```
...rubocop-1.40.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:308:in `department_marker?'
...rubocop-1.40.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:287:in `describe'
...rubocop-1.40.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:229:in `block in add_offense_for_entire_comment'
...rubocop-1.40.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:229:in `map'
...rubocop-1.40.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:229:in `add_offense_for_entire_comment'
...rubocop-1.40.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:220:in `block in add_offenses'
...rubocop-1.40.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:218:in `each'
...rubocop-1.40.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:218:in `add_offenses'
...rubocop-1.40.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:51:in `on_new_investigation'
```

The reason was the following:

```rb
    # causes an error
    def something
      # rubocop:disable Rails::SkipsModelValidations
      update_attribute(:rated, true)
      # rubocop:enable Rails::SkipsModelValidations
    end
    
    # works
    def something
      # rubocop:disable Rails/SkipsModelValidations
      update_attribute(:rated, true)
      # rubocop:enable Rails/SkipsModelValidations
    end
```

I don't know if the error was, that rubocop accepted the `Rails::SkipsModelValidations` or that the department-marker with `Rails::SkipsModelValidations` worked?

**Comment by pirj** [comment]
Thanks for reporting. I don't think it's related in any way to `rubocop-rspec` specifically, but it will certainly be an interesting case for [the `rubocop` team](https://github.com/rubocop/rubocop/issues/new/choose).
I don't have permissions to transfer this issue there, so I suggest you just reopen it there.

**Comment by koic** [comment]
> rubocop-1.40.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:308:in `department_marker?'

It may have been resolved in the latest RuboCop 1.43. If not, please open to the above.
